e2e_tests/tests/command/test_shell.py

```diff
@@ -43,3 +43,55 @@ def test_open_shell() -> None:
         child.read()
         child.wait()
         assert child.exitstatus == 0
+
+
+@pytest.mark.e2e_cpu
+def test_user_flag_shell() -> None:
+    with cmd.interactive_command("shell", "start", "--detach") as shell:
+        admin_username, admin_password = ADMIN_CREDENTIALS
+        default_empty = ""
+        task_id = shell.task_id
+        assert task_id is not None
+
+        child = det_spawn(["shell", "open", task_id])
+        child.setecho(True)
+        child.expect(r".*Permanently added.+([0-9a-f-]{36}).+known hosts\.", timeout=180)
+        child.sendline(f"det -u {admin_username} user logout")
+
+        # Use the -u flag with logged out admin user and verify password prompt
+        child.sendline(f"det -u {admin_username} user whoami")
+        child.expect(f"Password for user '{admin_username}'", timeout=10)
+        child.sendline(str(admin_password))
+        child.expect(f"You are logged in as user '{admin_username}'", timeout=10)
+
+        # Log into admin, create new user newusr with admin account, login and validate user
+        # identity
+        new_usr = "newusr"
```

Comment:
We have some machinery (`logged_in_user()`) for setting up users on the cluster, can you use that to create the temp user, rather than doing it interactively here?  It should be way less code, and it won't be subject to failures if you run the same test twice against the same cluster.

Reply:
Gotchya, incorporated this